suite-native/tokens/src/tokensSelectors.ts

```diff
@@ -1,70 +1,249 @@
+};
+
+export const selectAnyOfTokensIsKnown = (
+    state: TokenDefinitionsRootState & AccountsRootState,
+    accountKey: AccountKey,
+): boolean => {
+    // It may be temping to reuse selectAccountsKnownTokens.length but this is faster
+    const tokens = selectAllAccountTokens(state, accountKey);
+    const account = selectAccountByKey(state, accountKey);
+
+    if (!account?.symbol) {
+        return false;
+    }
+    const result = A.any(tokens, token => {
+        const isKnown = selectIsSpecificCoinDefinitionKnown(state, account.symbol, token.contract);
 
-    switch (coin) {
-        case 'eth':
-            const hasAnyTokens = selectHasDeviceAnyEthereumTokens(state);
+        return isKnown;
+    });
 
-            return hasAnyTokens;
-        default:
-            // Exhaustive check, all coin types in NETWORKS_WITH_TOKENS should be handled above
-            coin satisfies never;
+    return result;
+};
 
-            return false;
+const isNotZeroAmountTransfer = (tokenTransfer: TokenTransfer) =>
+    tokenTransfer.amount !== '' && tokenTransfer.amount !== '0';
+
+/** Is not a transaction with zero amount and no internal transfers. */
+const isNotEmptyTransaction = (transaction: WalletAccountTransaction) =>
+    transaction.amount !== '0' ||
+    (G.isArray(transaction.internalTransfers) && A.isNotEmpty(transaction.internalTransfers));
+
+const isTransactionWithTokenTransfers = (transaction: WalletAccountTransaction) =>
+    G.isArray(transaction.tokens) && A.isNotEmpty(transaction.tokens);
+
+const selectAccountTransactionsWithTokensWithFiatRates = memoizeWithArgs(
+    (
+        state: TransactionsRootState & TokenDefinitionsRootState,
+        accountKey: AccountKey,
+        areTokenOnlyTransactionsIncluded: boolean,
+    ): WalletAccountTransaction[] =>
+        pipe(
+            selectAccountTransactions(state, accountKey),
+            A.map(transaction => ({
+                ...transaction,
+                tokens: pipe(
+                    transaction?.tokens ?? [],
+                    A.filter(isNotZeroAmountTransfer),
+                    A.filter(token =>
+                        selectIsSpecificCoinDefinitionKnown(
+                            state,
+                            transaction.symbol,
+                            token.contract as TokenAddress,
+                        ),
+                    ),
+                    A.map((tokenTransfer: TokenTransfer) => ({
+                        ...tokenTransfer,
+                        symbol: tokenTransfer.symbol,
+                    })),
+                ) as TypedTokenTransfer[],
+            })),
+            A.filter(
+                transaction =>
+                    isNotEmptyTransaction(transaction) ||
+                    (areTokenOnlyTransactionsIncluded &&
+                        isTransactionWithTokenTransfers(transaction)),
+            ),
+        ) as WalletAccountTransaction[],
+    { size: 500 },
```

Comment:
I don't think there will be 500 accounts

Reply:
e8dde2170e